src/components/themed/PromoBanner.js

```diff
@@ -14,6 +17,51 @@ export const PromoBanner = () => {
 
   const accountMessages = useSelector(state => state.account.referralCache.accountMessages)
   const accountReferral = useSelector(state => state.account.accountReferral)
+  const store = useSelector(state => state.core.account.dataStore)
+
+  const [promotions, setPromotions] = useState<Promotion[]>([])
+  const [localFlags, setLocalFlags] = useState<AppFlags>({})
+
+  // Define user profile
+  const profile = useMemo(() => getProfile(), [])
+
+  // Get active promotions
+  useEffect(() => {
+    let abort = false
+    getPromotions().then(promos => {
+      if (abort) return
+      setPromotions(promos.filter(promo => testProfile(profile, promo)))
+    })
+
+    return () => {
+      abort = true
+    }
+  }, [profile])
+
+  // Test for wyreLinkedBank
+  useEffect(() => {
```

Comment:
Use the 'useAsyncEffect' hook we have which already checks for safety and catches errors.